K2Bridge/Visitors/RangeVisitor.cs

```diff
@@ -0,0 +1,72 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license.
+// See LICENSE file in the project root for full license information.
+
+namespace K2Bridge.Visitors
+{
+    using System.Globalization;
+    using K2Bridge.Models.Request.Aggregations;
+    using K2Bridge.Models.Response;
+
+    /// <content>
+    /// A visitor for the <see cref="RangeAggregation"/> element.
+    /// </content>
+    internal partial class ElasticSearchDSLVisitor : IVisitor
+    {
+        /// <inheritdoc/>
+        public void Visit(RangeAggregation rangeAggregation)
+        {
+            Ensure.IsNotNull(rangeAggregation, nameof(RangeAggregation));
+            EnsureClause.StringIsNotNullOrEmpty(rangeAggregation.Metric, nameof(RangeAggregation.Metric));
+            EnsureClause.StringIsNotNullOrEmpty(rangeAggregation.Field, nameof(RangeAggregation.Field));
+
+            // Start the union operator
+            rangeAggregation.KustoQL += "union ";
+
+            // Query expressions for each range
+            // (_data | where foo >= 1 and bar < 10 | summarize ['3']=avg(baz), count() | extend ['2'] = '1-10')
+            //          ^^^^^^^^^^^^^^^^^^^^^^^^^^^
+            //          The range clause
+            //                                        ^^^^^^^^^^^^^^^^^^^^^^^^
+            //                                        The metrics expressions
+            //                                                                  ^^^^^^^
+            //                                                                  Default metric count()
+            //                                                                            ^^^^^^^^^^^^^^^^^^^^^
+            //                                                                            Column with the range name
+            foreach (var range in rangeAggregation.Ranges)
+            {
+                rangeAggregation.KustoQL += $"(_data | {KustoQLOperators.Where} ";
+
+                range.Field = rangeAggregation.Field;
+                range.Accept(this);
+
+                // Insert the range clause
+                if (string.IsNullOrEmpty(range.KustoQL))
+                {
+                    // This is then "open" range, -infinity to +infinity
+                    rangeAggregation.KustoQL += "true";
```

Comment:
really needed? why not 'continue'?

Reply:
Good catch, thanks! The if statement can be simplified and the query too.